src/transit_equity/census/income_poverty_level.py

```diff
@@ -0,0 +1,252 @@
+    with 100 people in total, 20 people with income to poverty level ratio less than 0.5,
+    30 people with income to poverty level ratio between 0.5 and 0.99, 
+    10 people with income to poverty level ratio between 1.0 and 1.24, 
+    40 people with income to poverty level ratio between 2.0 and above, 
+    and 0 people with income to poverty level ratio in between.
+
+    The income to poverty level ratio range is [0, 2.0).
+    
+    Expected Answer: (20 + 30 + 10) = 60
+
+    >>> import pandas as pd
+    >>> from transit_equity.census.income_poverty_level import INCOME_POVERTY_LEVEL_COLUMNS
+    >>> from transit_equity.census.income_poverty_level import get_population_in_income_poverty_level_range
+    >>> income_poverty_level_dict = {
+    ...     'C17002_001E': 100,
+    ...     'C17002_002E': 20,
+    ...     'C17002_003E': 30,
+    ...     'C17002_004E': 10,
+    ...     'C17002_005E': 0,
+    ...     'C17002_006E': 0,
+    ...     'C17002_007E': 0,
+    ...     'C17002_008E': 40,
+    }
+    >>> income_poverty_level_row = pd.Series(income_poverty_level_dict)
+    >>> get_population_in_income_poverty_level_range(income_poverty_level_row, 0, 2.0)
+    60
+    """
+    population = 0
+    true_min_income_poverty_level = min_income_poverty_level
+    true_max_income_poverty_level = max_income_poverty_level
+    for column in INCOME_POVERTY_LEVEL_COLUMNS:
+        if column.value.field not in income_poverty_level_row:
+            continue
+        # Check if the value of the field is within [min_income_poverty_level, max_income_poverty_level]
+        if column.value.min_income_poverty_level >= min_income_poverty_level and column.value.max_income_poverty_level < max_income_poverty_level:
+            population += income_poverty_level_row[column.value.field]
+        # Else check if the value of the field has no overlap with [min_income_poverty_level, max_income_poverty_level]
+        elif column.value.min_income_poverty_level >= max_income_poverty_level or column.value.max_income_poverty_level < min_income_poverty_level:
+            pass
+        # Else there is some overlap with [min_income_poverty_level, max_income_poverty_level)
+        else:
+            population += income_poverty_level_row[column.value.field]
+            if column.value.min_income_poverty_level < min_income_poverty_level:
+                true_min_income_poverty_level = column.value.min_income_poverty_level
+            if column.value.max_income_poverty_level > max_income_poverty_level:
+                true_max_income_poverty_level = column.value.max_income_poverty_level
+    return {
+        'population': population,
+        'true_min_income_poverty_level': true_min_income_poverty_level,
+        'true_max_income_poverty_level': true_max_income_poverty_level
+    }
+
+def get_population_in_income_poverty_level_range_df(income_poverty_level_row_df: pd.DataFrame,
```

Comment:
@bhazelton 
I have added a new method that processes the entire dataframe, instead of a single row. I added this because the functions are now returning dictionaries/tuples instead of single values.
There may be a requirement to refactor all of this, so that the entire module is more cohesive. But at the moment, imo, we should have both these methods for my work.
I guess one criticism you would have is the parameters (and return keys) have quite large names.

Reply:
Regarding the variable names, I see that I am adding some redundancy in names, that don't seem to help much imo.
Maybe I can remove the redundancy:
E.g.: max_income_poverty_level -> max_level

Reply:
This new method seems very reasonable, is there still a need for the one that operates on just one row? Can't you just pass a df with one row to the new one if you only want one row's worth?

Reply:
That is very fair actually. Let me get back on this.

Reply:
I think your ideas about variable names are sensible, just make sure they're descriptive enough to follow.

Reply:
Alright then. 
I'll be getting rid of the function that processes Series object.

Reply:
Hello @bhazelton ,
I have gotten rid of the series based method.
I am still working on the variable names. I just realized that the variable name change I suggested may not make the most sense:
min_level is weird since that may mislead people into thinking that the denominator varies, when it is actually the ratio that varies.

Reply:
I should instead do the following naming convention.
Instead of min_income_poverty_level, I can say min_ratio, since that's is what the variable actually indicates.

Reply:
So @bhazelton ,
I have changed the naming convention quite a lot. I realize that not having the word ratio felt a little confusing for even me, let alone someone who hasn't looked into my code.
I hope this naming convention is fine.
I have tested this code in the private notebook, and it is all working fine. 
Please do let me know what do you think about this one.

I don't think I have any other changes proposed for this PR.
Thank you!